sys/net/routing/rpl/rpl_control_messages.c

```diff
@@ -252,9 +248,18 @@ void rpl_init_root(rpl_options_t *rpl_opts)
 
 }
 
+/* TODO: this function is used by the ip layer for source routing (non-storing mode)
+ * to determine if the node is root and change the dodag direction (upwards -> downwards).
+ * This is a hack and needs to be refactored when the new network stack is ready.
+ */
 uint8_t rpl_is_root(void)
 {
-    return i_am_root;
+    for (uint8_t i = 0; i < RPL_MAX_DODAGS; i++) {
+        if (rpl_dodags[i].node_status == ROOT_NODE) {
```

Comment:
@cgundogan what do you think could we use `rpl_dodags[i].my_rank == RPL_ROOT_RANK` to determine if this node is root and remove the `rpl_dodag_t::node_status` member?